lib/wraft_doc/search/typesense.ex

```diff
@@ -0,0 +1,41 @@
+defmodule WraftDoc.Search.Typesense do
+  @moduledoc """
+  Handles operations for Typesense
+  """
+
+  alias WraftDoc.Search.Encoder
+
+  def create_collection(schema) do
+    ExTypesense.create_collection(schema)
+  end
+
+  def create_document(document) do
+    document = Encoder.to_document(document)
+    ExTypesense.create_document(document)
+  end
+
+  def get_document(id, collection_name) do
+    ExTypesense.get_document(collection_name, to_string(id))
+  end
+
+  def update_document(document) do
+    document = Encoder.to_document(document)
+    ExTypesense.upsert_document(document)
+  end
+
+  def delete_document(id, collection_name) do
+    query = %{filter_by: "id: #{id}"}
```

Comment:
Since we are deleting only a single record, do we want to be using `ExTypesense.delete_documents_by_query/2`? Can't we just use `ExTypesense.delete_record/1`

Also, the query here is wrong. https://typesense.org/docs/26.0/api/search.html#filter-parameters 
Check out the documentation